synapse/replication/tcp/client.py

```diff
@@ -152,6 +152,7 @@ async def on_rdata(
             rows: a list of Stream.ROW_TYPE objects as returned by Stream.parse_row.
         """
         self.store.process_replication_rows(stream_name, instance_name, token, rows)
+        self.store.process_replication_position(stream_name, instance_name, token)
```

Comment:
Can you add a quick comment here to say that `process_replication_position` *must* be called after `process_replication_rows` please? Just so that anyone looking at this function knows that this is a bit sensitive.

Reply:
https://github.com/matrix-org/synapse/pull/14723/commits/537cb404678f050db9d00262bdeb86bee6dfba67